### ISSUE TITLE
Authors with 2 family names not formatted correctly by `create_epidist_citation()`

### ISSUE THREAD

**Issue by joshwlambert** [epiverse-trace/epiparameter#154]
I think this will cause issues for authors for 2 family names. Is it possible to do less formatting here?

_Originally posted by @Bisaloo in https://github.com/epiverse-trace/epiparameter/pull/151#discussion_r1202773402_

**Comment by joshwlambert** [comment]
This issue still persists if the author's 2 family names are separated by a space:

``` r
library(epiparameter)
create_epidist_citation(
  author = "Smith-Rowe", 
  year = 2000, 
  title = "A paper", 
  journal = "Epi Journal", 
  DOI = "10.37125326ha"
)
#> Using Smith-Rowe (2000). "A paper." _Epi Journal_. doi:10.37125326ha
#> <https://doi.org/10.37125326ha>. 
#> To retrieve the citation use the 'get_citation' function
#> Smith-Rowe (2000). "A paper." _Epi Journal_. doi:10.37125326ha
#> <https://doi.org/10.37125326ha>.

create_epidist_citation(
  author = "Smith Rowe", 
  year = 2000, 
  title = "A paper", 
  journal = "Epi Journal", 
  DOI = "10.37125326ha"
)
#> Using Rowe S (2000). "A paper." _Epi Journal_. doi:10.37125326ha
#> <https://doi.org/10.37125326ha>. 
#> To retrieve the citation use the 'get_citation' function
#> Rowe S (2000). "A paper." _Epi Journal_. doi:10.37125326ha
#> <https://doi.org/10.37125326ha>.
```

<sup>Created on 2023-08-29 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

This issue will be resolved when the {epiparameter} library is updated to include all author's names for each entry.

**Comment by joshwlambert** [comment]
This is now (mostly) resolved with the changes merged in #212. On the database side this is fully taken care of and should no longer be an issue. 

When interactively creating `<epidist>` citations this is mostly solved by specifying a `<person>` object. If a `character` is given then the formatting follows `as.person()` which is not perfect and does not always work as wanted for 2 family names (see last example). As using the `<person>` class is the recommended method and the secondary option uses the base R standard I'm happy to consider this issue resolved.

``` r
library(epiparameter)

create_epidist_citation(
  author = person(given = "John", family = "Smith-Rowe"), 
  year = 2022, 
  title = "A paper", 
  journal = "A journal", 
  DOI = "10.2326328x"
)
#> Using Smith-Rowe J (2022). "A paper." _A journal_. doi:10.2326328x
#> <https://doi.org/10.2326328x>. 
#> To retrieve the citation use the 'get_citation' function
#> Smith-Rowe J (2022). "A paper." _A journal_. doi:10.2326328x
#> <https://doi.org/10.2326328x>.

create_epidist_citation(
  author = person(given = "John", family = "Smith Rowe"), 
  year = 2022, 
  title = "A paper", 
  journal = "A journal", 
  DOI = "10.2326328x"
)
#> Using Smith Rowe J (2022). "A paper." _A journal_. doi:10.2326328x
#> <https://doi.org/10.2326328x>. 
#> To retrieve the citation use the 'get_citation' function
#> Smith Rowe J (2022). "A paper." _A journal_. doi:10.2326328x
#> <https://doi.org/10.2326328x>.

create_epidist_citation(
  author = c("John Smith Rowe"), 
  year = 2022, 
  title = "A paper", 
  journal = "A journal", 
  DOI = "10.2326328x"
)
#> Using Rowe JS (2022). "A paper." _A journal_. doi:10.2326328x
#> <https://doi.org/10.2326328x>. 
#> To retrieve the citation use the 'get_citation' function
#> Rowe JS (2022). "A paper." _A journal_. doi:10.2326328x
#> <https://doi.org/10.2326328x>.
```

<sup>Created on 2023-11-28 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>
